source/agora/utils/Test.d

```diff
@@ -677,6 +677,25 @@ public struct TxBuilder
         return this;
     }
 
+    /***************************************************************************
+
+        Deduct all the remaining amount, so there will be no refund transaction
+
+        Useful when trying to test fee calculation
+
+        Returns:
+            Reference to `this` for easy chaining
+
+    ***************************************************************************/
+
+    public ref typeof(this) deductRemaining ()
+        return
+    {
+        this.leftover.value = Amount(0);
+
+        return this;
+    }
+
```

Comment:
Nope, this is one of those function that sounds good but will be a pain down the road.
The `TxBuilder` was made so that valid transactions are easy to make and invalid ones are hard.
This function does not, as it makes it trivial to create transactions that burns an arbitrary amount, something we seldom see IRL.

Reply:
I don't thing burning would make the transaction invalid. :) The suggested usecase for this function is txbuilder.draw(100, [pk]).deductRemaining() which essentially set the remaining amount as fee. I can make this function version(unittest) if you would like it.

We already have a deduct() function, so alternatively I could have defined a new function, that returned the remaining amount and pass it to deduct.